### DISCUSSION TITLE
Prisma extends - Adding extra types to the args for current methods

### DISCUSSION THREAD

**Discussion by MilesNorton** [prisma/prisma#18251] [Q&A] [answered]
I'm trying to enforce passing additional args to the current prisma methods on models so that I can intercept the request and then audit requests against the additional args (userId).

I was just wondering if someone could help me because I'm using the advanced types [docs](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/model#advanced-type-safety-improve-the-type-safety-and-developer-experience-of-your-custom-model-methods) to do this but when I pass 'selects' and 'includes' the return type doesn't enforce the select and include types.

Here is an example: 
```ts
const prisma = new PrismaClient();

const xprisma = prisma.$extends({
  model: {
    post: {
      async upsert<T, A>(
        this: T,
        x: Prisma.Exact<
          A,
          Prisma.Args<T, "findFirst"> & { userId: string }
        >
  ): Prisma.Result<T, A, "findFirst"> {
        await prisma.audit.create({
          userId: x.userId,
          // ... extra fields
        });

        delete x.userId;

        return await prisma.post.upsert(x);
      },
    },
  },
});

async function main() {
  const test0 = xprisma.post.upsert({
    userId: 'uuid',
	
    select: {
      id: true,
    },
  });
// test0 isn't typed to just id.
}
```

**Comment by nurul3101** [comment]
Hello @MilesNorton 👋 

Can you share the post and audit models as well? 
The ideal outcome should be that return type of upsert function should only have `id` field, right?

**Reply by nurul3101** [reply]
@MilesNorton Did you had a chance to look at my response? Please let us know in case you still need assistance.

**Reply by MilesNorton** [reply, marked as answer]
Hey mate, sorry. We abandoned this route and instead used db hooks to enforce field changes
